### ISSUE TITLE
Incorrect error 2301 when using ES standard class properties

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#49719]
As explained in https://github.com/Microsoft/TypeScript/issues/2548, the following code is not correct when `useDefineForClassFields: false`, the default for targets before ES2021, or when target <= ES2020.

```ts
const x = 1
class C {
    p = x
    constructor(x: string) { }
}
```

However, when `useDefineForClassFields: true` and target >= ES2021, there shouldn't be an error here.

Expected: No error 2301 on `private p = x`, and `p: number`
Actual: Error on `private p = x` and `p: any`

Edit: @fatcerberus pointed out that the error is only incorrect when `useDefineForClassFields: true` and target >= ES2021, because downlevel define for class fields still happens in the constructor.

**Comment by fatcerberus** [comment]
This is broken because the outer `x` gets shadowed by the ctor parameter, right?  But that means it’s also broken for downleveled `useDefineForClassFields` too…
